crates/vm/levm/src/vm.rs

```diff
@@ -1303,6 +1371,226 @@ impl VM {
+        }
+
+        let (code_address_info, _) = self.access_account(initial_call_frame.code_address);
+
+        if has_delegation(&code_address_info)? {
+            initial_call_frame.code_address = get_authorized_address(&code_address_info)?;
+            let (auth_address_info, _) = self.access_account(initial_call_frame.code_address);
+
+            initial_call_frame.bytecode = auth_address_info.bytecode.clone();
+        } else {
+            initial_call_frame.bytecode = code_address_info.bytecode.clone();
+        }
+
+        initial_call_frame.valid_jump_destinations =
+            get_valid_jump_destinations(&initial_call_frame.bytecode).unwrap_or_default();
+        Ok(refunded_gas)
+    }
+
+    /// Used for the opcodes
+    /// The following reading instructions are impacted:
+    ///      EXTCODESIZE, EXTCODECOPY, EXTCODEHASH
+    /// and the following executing instructions are impacted:
+    ///      CALL, CALLCODE, STATICCALL, DELEGATECALL
+    /// In case a delegation designator points to another designator,
+    /// creating a potential chain or loop of designators,
+    /// clients must retrieve only the first code and then stop following the designator chain.
+    ///
+    /// For example,
+    /// EXTCODESIZE would return 2 (the size of 0xef01) instead of 23 which would represent the delegation designation,
+    /// EXTCODEHASH would return 0xeadcdba66a79ab5dce91622d1d75c8cff5cff0b96944c3bf1072cd08ce018329 (keccak256(0xef01)), and
+    /// CALL would load the code from address and execute it in the context of authority.
+    ///
+    /// The idea of this function comes from ethereum/execution-specs:
+    /// https://github.com/ethereum/execution-specs/blob/951fc43a709b493f27418a8e57d2d6f3608cef84/src/ethereum/prague/vm/eoa_delegation.py#L115
+    pub fn eip7702_get_code(
+        &mut self,
+        address: Address,
+    ) -> Result<(bool, u64, Address, Bytes), VMError> {
+        // Address is the delgated address
+        let account = get_account(&mut self.cache, &self.db, address);
+        let bytecode = account.info.bytecode.clone();
+
+        // If the Address doesn't have a delegation code
+        // return false meaning that is not a delegation
+        // return the same address given
+        // return the bytecode of the given address
+        if !has_delegation(&account.info)? {
+            return Ok((false, 0, address, bytecode));
+        }
+
+        // Here the address has a delegation code
+        // The delegation code has the authorized address
+        let auth_address = get_authorized_address(&account.info)?;
+
+        let access_cost = match self.accrued_substate.touched_accounts.get(&auth_address) {
+            Some(_) => WARM_ADDRESS_ACCESS_COST,
+            None => {
+                self.accrued_substate.touched_accounts.insert(auth_address);
+                COLD_ADDRESS_ACCESS_COST
+            }
+        };
```

Comment:
We do not need to get the account if we are not going to use it.
```suggestion
        let access_cost = if self
            .accrued_substate
            .touched_accounts
            .contains(&auth_address)
        {
            WARM_ADDRESS_ACCESS_COST
        } else {
            self.accrued_substate.touched_accounts.insert(auth_address);
            COLD_ADDRESS_ACCESS_COST
        };
```